### PR TITLE
Change so that both user and group id validators will work

### PR DESCRIPTION
**Pull request by VladislavStefanov**
Signed-off-by: Vladislav Stefanov <vladcata22@abv.bg>

### CODE REVIEW

File: scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
+    this.uriInfo = uriInfo;
 
-  public IdValidator(@Context Application application) {
-    this.customValidators = SCIMApplication.from(application).getCustomConstraintValidators();
   }
 
   @Override
   public boolean isValid(Object resourceId, ConstraintValidatorContext constraintValidatorContext) {
-    if(!customValidators.containsKey(RESOURCE_ID) || customValidators.get(RESOURCE_ID).isValid(resourceId)) {
+    if (uriInfo.getPath().startsWith(API.GROUPS) && customValidators.containsKey(GROUP_ID)) {
```
Comment (hborisov) [inline scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java]: Looks good, but let's see if we can avoid the nested if statements.

File: scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
+
+    if (uriInfo.getPath().startsWith(API.USERS) && customValidators.containsKey(USER_ID)) {
+      if (customValidators.get(USER_ID).isValid(resourceId)) {
+        return true;
+      }
+      ValidationUtil.interpolateErrorMessage(constraintValidatorContext, generateViolationMessage(resourceId));
+      return false;
+    }
+
+    if (!customValidators.containsKey(RESOURCE_ID) || customValidators.get(RESOURCE_ID).isValid(resourceId)) {
```
Comment (hborisov) [inline scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java]: I am not sure if this check is correct. If there is no resource id validator then validation should pass sucessfully?!

File: scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java (hunk 1 above)
Comment (VladislavStefanov) [reply scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java]: Then the two nested if statements will become 4 if statements.
Should I change it?

Comment (hborisov) [reply scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java]: Maybe sth like this:

`@Override
  public boolean isValid(Object resourceId, ConstraintValidatorContext constraintValidatorContext) {
    if (customValidators.isEmpty()) {
      return true;
    }

    if (uriInfo.getPath().startsWith(API.GROUPS) && customValidators.containsKey(GROUP_ID) && customValidators.get(GROUP_ID).isValid(resourceId)) {
      return true;
    }

    if (uriInfo.getPath().startsWith(API.USERS) && customValidators.containsKey(USER_ID) && customValidators.get(USER_ID).isValid(resourceId)) {
      return true;
    }

    if (customValidators.containsKey(RESOURCE_ID) && customValidators.get(RESOURCE_ID).isValid(resourceId)) {
      return true;
    }

    ValidationUtil.interpolateErrorMessage(constraintValidatorContext, generateViolationMessage(resourceId));
    return false;
  }`

Comment (hborisov) [reply scimono-server/src/main/java/com/sap/scimono/entity/schema/validation/IdValidator.java]: Let's discuss it tomorrow

**Pull request merged**